### PR TITLE
Remove wallet with confirm dialog

### PR DESCRIPTION
**Pull request by sergous**
Show delete button for API wallets with empty total balance.
On click delete show confirm window.

### CODE REVIEW

File: src/components/EditWalletDrawer/index.tsx [hunk 1]
```diff
@@ -1,40 +1,112 @@
...
 import Drawer from '../Drawer';
 import EditWalletForm from '../EditWalletForm/index';
+import {Icon} from '../Icon/index';
+import {Link} from '../Link/index';
 import {Wizard, WizardStep} from '../Wizard';
 
 interface EditWalletDrawerProps {
   wallet?: WalletModel;
   show?: boolean;
+  toggleConfirmRemoveWallet?: () => void;
```
Comment (optimusway) [inline src/components/EditWalletDrawer/index.tsx]: i don't think `toggleConfirmRemoveWallet` and `showConfirmRemoveWallet` should be exposed in component api

File: src/components/EditWalletDrawer/index.tsx [hunk 2]
```diff
@@ -1,40 +1,112 @@
...
+    return (
+      wallet && (
+        <Drawer title="Edit Wallet" show={show}>
+          <div className="drawer__title">
+            <h2>{wallet.title}</h2>
+            <h3>API Wallet</h3>
+            {wallet.isDeletable && (
+              <div className="pull-right">
+                <Link
+                  // tslint:disable-next-line:jsx-no-lambda
```
Comment (optimusway) [inline src/components/EditWalletDrawer/index.tsx]: there is no lambda below, plz remove

File: src/components/EditWalletDrawer/index.tsx [hunk 3]
```diff
@@ -1,40 +1,112 @@
...
+                  <Icon name="recycle" />
+                  Delete wallet
+                </Link>
+              </div>
+            )}
+          </div>
+          <Wizard activeIndex={1}>
+            <WizardStep
+              title="Name and description"
+              onCancel={null}
```
Comment (optimusway) [inline src/components/EditWalletDrawer/index.tsx]: `={null}` is useless, plz remove

File: src/components/EditWalletDrawer/index.tsx [hunk 4]
```diff
@@ -1,40 +1,112 @@
...
-}))(observer(EditWalletDrawer));
+  private toggleConfirm = () =>
+    this.props.toggleConfirmRemoveWallet &&
+    this.props.toggleConfirmRemoveWallet();
+
+  private handleRemoveWallet = () => {
+    this.toggleConfirm();
+    const {onRemoveWallet, wallet} = this.props;
+    if (onRemoveWallet && wallet) {
+      onRemoveWallet(wallet!);
```
Comment (optimusway) [inline src/components/EditWalletDrawer/index.tsx]: no need for bang operator `!` inside conditional statement, if not mistaken

File: src/stores/walletStore.spec.ts [hunk 5]
```diff
@@ -42,6 +50,36 @@ describe('wallet store', () => {
...
   });
 
+  describe('existingWallet', () => {
+    let wallet: WalletModel;
+
+    beforeEach(async () => {
+      walletStore.clearWallets();
+      const dto = newWalletApiKey();
+      const store = walletStore;
+      wallet = new WalletModel(store, dto);
```
Comment (optimusway) [inline src/stores/walletStore.spec.ts]: as a side note: it's better to instantiate objects with the store, `walletStore.createWallet` in this case

File: src/models/walletModel.ts [hunk 6]
```diff
@@ -22,6 +22,11 @@ export class WalletModel {
     return this.balances.map(b => b.balanceInBaseAsset).reduce(sum, 0);
   }
 
+  @computed
+  get isDeletable() {
+    return !this.isTrading && this.totalBalance === 0;
```
Comment (optimusway) [inline src/models/walletModel.ts]: probably `&& !!this.id` is needed

Comment (optimusway) [reply src/models/walletModel.ts]: cover this by tests plz

File: src/stores/walletStore.ts [hunk 7]
```diff
@@ -82,6 +87,17 @@ export class WalletStore {
...
   };
 
+  removeApiWallet = async (wallet: WalletModel) => {
+    const {id} = wallet;
+    try {
+      return await this.api!.removeApiWallet(id);
+    } catch (reason) {
+      return Promise.reject(reason);
+    } finally {
+      this.removeWallet(wallet);
```
Comment (optimusway) [inline src/stores/walletStore.ts]: but you _don't need to remove_ the wallet from the list in a case of failure, plz add a test for that

File: src/components/EditWalletDrawer/index.tsx (hunk 1 above)
Comment (sergous) [reply src/components/EditWalletDrawer/index.tsx]: agreed, I will remove them

File: src/components/EditWalletDrawer/index.tsx (hunk 2 above)
Comment (sergous) [reply src/components/EditWalletDrawer/index.tsx]: ok

File: src/components/EditWalletDrawer/index.tsx (hunk 3 above)
Comment (sergous) [reply src/components/EditWalletDrawer/index.tsx]: ok

File: src/stores/walletStore.ts (hunk 7 above)
Comment (sergous) [reply src/stores/walletStore.ts]: yes, also I will show error in uiStore

File: src/stores/walletStore.spec.ts (hunk 5 above)
Comment (sergous) [reply src/stores/walletStore.spec.ts]: agree

File: src/models/walletModel.ts (hunk 6 above)
Comment (sergous) [reply src/models/walletModel.ts]: will do

**Comment by sergous** [comment]
Requested changes uploaded @optimusway

**Comment by sergous** [comment]
Rebased, please review @optimusway

**Pull request closed without merging**